src/main/scala/Core/Top.scala

```diff
@@ -7,23 +7,20 @@ import chisel3.util.experimental.loadMemoryFromFileInline // Load the contents o
 abstract class AbstractTop(cfg: FlexpretConfiguration) extends Module {
 
     // Write flexpret_config.h and flexpret_config.ld to file
-    cfg.writeConfigHeaderToFile("programs/lib/include/flexpret_config.h")
+    cfg.writeConfigHeaderToFile("programs/lib/include/flexpret_hwconfig.h")
     cfg.writeLinkerConfigToFile("programs/lib/linker/flexpret_config.ld")
+    cfg.writeMakeConfigToFile("./config.mk")
```

Comment:
Should these be parametrized? or at the least documented well given hardcoded paths and constants

Reply:
I added some explanations for it on the last commit. 

I'm not quite sure how it would be best parametrized. If, say, we parametrize the path for flexpret_hwconfig.h and the user changes its path to be generated to ./programs/lib/flexpret_hwconfig.h (as opposed to ./programs/lib/include/flexpret_hwconfig.h), then the software build system would somehow need to be aware of this new path as well.

Also, I don't quite see other locations where one would want to place these files apart from these hardcoded paths.